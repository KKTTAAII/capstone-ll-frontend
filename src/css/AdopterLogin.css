.AdopterLogin-container {
    text-align: center;
    margin-top: 10%;
}

.AdopterLogin-header {
    font-size: 2em;
    margin: 20px;
}

.AdopterLogin-form {
    display: flex;
    flex-direction: column;
    width: 300px;
    align-items: center;
    justify-content: center;
    margin: 0 auto;
}

.AdopterLogin-label {
    font-weight: 400;
    text-transform: uppercase;
    margin-bottom: 5px;
}

.AdopterLogin-input {
    width: 400px;
    background-color: transparent;
    border: none;
    color: black;
    outline: none;
    border-bottom: 1px solid #000000;
    font-size: 1em;
    font-weight: 300;
    padding-bottom: 10px;
    margin-top: 10px;
    margin-bottom: 10px;
}

.AdopterLogin-input:focus {
    box-shadow: rgba(0, 0, 0, 0.1) 0px 10px 15px -3px, rgba(0, 0, 0, 0.05) 0px 4px 6px -2px;
}

.AdopterLogin-warning {
    font-size: x-small;
    color: red;
    margin: 10px;
    font-weight: 400;
}

.AdopterLogin-button {
    border-radius: 10px;
    border-style: none;
    background-color: white;
    height: 40px;
    width: 70px;
    margin: 15px;
    box-shadow: rgba(0, 0, 0, 0.25) 0px 54px 55px, rgba(0, 0, 0, 0.12) 0px -12px 30px, rgba(0, 0, 0, 0.12) 0px 4px 6px, rgba(0, 0, 0, 0.17) 0px 12px 13px, rgba(0, 0, 0, 0.09) 0px -3px 5px;
}

.AdopterLogin-button:hover {
    box-shadow: rgba(0, 0, 0, 0.07) 0px 1px 2px, rgba(0, 0, 0, 0.07) 0px 2px 4px, rgba(0, 0, 0, 0.07) 0px 4px 8px, rgba(0, 0, 0, 0.07) 0px 8px 16px, rgba(0, 0, 0, 0.07) 0px 16px 32px, rgba(0, 0, 0, 0.07) 0px 32px 64px;
}

.AdopterLogin-reminder {
    margin-top: 20px;
}

.AdopterLogin-signuplink {
    text-decoration: none;
    color: rgb(226, 129, 148);
}