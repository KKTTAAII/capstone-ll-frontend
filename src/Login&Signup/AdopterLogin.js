import React, { useState } from "react";
import { useHistory, Link } from "react-router-dom";
import swal from "sweetalert";
import { WARNING, checkAllRequiredField, createInput } from "../common/helpers";
import "../css/AdopterLogin.css";

const INITIAL_STATE = {
  username: "",
  password: "",
};

const AdopterLogin = ({ logIn }) => {
  const [formData, setFormData] = useState(INITIAL_STATE);
  const [isTouched, setIsTouched] = useState(false);
  const [isInvalid, setIsInvalid] = useState(true);
  const history = useHistory();

  const handleChange = e => {
    const { name, value } = e.target;
    setFormData(formData => ({
      ...formData,
      [name]: value,
    }));
    setIsTouched(true);
    e.target.value === "" ? setIsInvalid(true) : setIsInvalid(false);
  };

  const handleSubmit = async e => {
    e.preventDefault();
    setFormData(INITIAL_STATE);
    const { username, password } = formData;
    const isAllRequiredFieldFilled = checkAllRequiredField([
      username,
      password,
    ]);
    if (!isInvalid && isAllRequiredFieldFilled) {
      const response = await logIn("Adopter", formData);
      //if there is a response, there is an error
      if (response) {
        swal({ text: response[0], icon: "warning" });
      } else {
        history.push("/adoptableDogs");
      }
    } else {
      swal({
        text: "Oops, please fill out all required fields",
        icon: "warning",
      });
      console.log("Oops, please fill out all required fields");
    }
  };

  return (
    <div className="AdopterLogin-container">
      <div className="AdopterLogin-header">Adopter Log in</div>
      <form onSubmit={handleSubmit} className="AdopterLogin-form">
        {createInput(
          "username",
          "text",
          formData.username,
          handleChange,
          "Username",
          false,
          "AdopterLogin-label",
          "AdopterLogin-input"
        )}
        {createInput(
          "password",
          "password",
          formData.password,
          handleChange,
          "Password",
          false,
          "AdopterLogin-label",
          "AdopterLogin-input"
        )}
        {isInvalid && isTouched && (
          <small className="AdopterLogin-warning">{WARNING}</small>
        )}
        <button className="AdopterLogin-button">Log in</button>
      </form>
      <div className="AdopterLogin-reminder">
        Don't have an account yet? Sign up{" "}
        <Link to={`/adopters/signup`} className="AdopterLogin-signuplink">
          here
        </Link>
      </div>
    </div>
  );
};

export default AdopterLogin;
