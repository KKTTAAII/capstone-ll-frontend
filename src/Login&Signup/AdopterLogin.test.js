import React from "react";
import { fireEvent, render } from "@testing-library/react";
import AdopterLogin from "./AdopterLogin";
import { MemoryRouter } from "react-router-dom";

it("renders without crashing", () => {
  render(
    <MemoryRouter>
      <AdopterLogin />
    </MemoryRouter>
  );
});

it("matches snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <AdopterLogin />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("shows log in button", () => {
  const { getByText } = render(
    <MemoryRouter>
      <AdopterLogin />
    </MemoryRouter>
  );

  //expect the Adopter Log in element in the Document
  expect(getByText("Adopter Log in")).toBeInTheDocument();
  expect(getByText("Log in")).toBeInTheDocument();
});

it("shows error if the form is submitted wihtout required fields", () => {
  const { getByText } = render(
    <MemoryRouter>
      <AdopterLogin />
    </MemoryRouter>
  );

  //click the log in button without filling out the form
  const logInButton = getByText("Log in");
  fireEvent.click(logInButton);

  //an error will pop up with a reminder
  expect(
    getByText("Oops, please fill out all required fields")
  ).toBeInTheDocument();
});


