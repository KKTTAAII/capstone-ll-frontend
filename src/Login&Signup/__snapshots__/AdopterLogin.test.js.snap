// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="AdopterLogin-container"
  >
    <div
      class="AdopterLogin-header"
    >
      Adopter Log in
    </div>
    <form
      class="AdopterLogin-form"
    >
      <label
        class="AdopterLogin-label"
      >
        Username:
      </label>
      <input
        class="AdopterLogin-input"
        id="username"
        name="username"
        placeholder="Username"
        type="text"
        value=""
      />
      <label
        class="AdopterLogin-label"
      >
        Password:
      </label>
      <input
        class="AdopterLogin-input"
        id="password"
        name="password"
        placeholder="Password"
        type="password"
        value=""
      />
      <button
        class="AdopterLogin-button"
      >
        Log in
      </button>
    </form>
    <div
      class="AdopterLogin-reminder"
    >
      Don't have an account yet? Sign up 
      <a
        class="AdopterLogin-signuplink"
        href="/adopters/signup"
      >
        here
      </a>
    </div>
  </div>
</DocumentFragment>
`;
